sbiexpt/bijectors.py

```diff
@@ -0,0 +1,101 @@
+import jax
+from jax import lax
+import jax.numpy as jnp
+from functools import partial
+import tensorflow_probability as tfp; tfp = tfp.substrates.jax
+tfd = tfp.distributions
+
+__all__ = ['ImplicitRampBijector']
+
+
+@partial(jax.jit, static_argnums=(0,))
+def fwd_solver(f, z_init):
+  def cond_fun(carry):
+    z_prev, z = carry
+    return jnp.linalg.norm(z_prev - z) > 1e-4
+
+  def body_fun(carry):
+    _, z = carry
+    return z, f(z)
+
+  init_carry = (z_init, f(z_init))
+  _, z_star = lax.while_loop(cond_fun, body_fun, init_carry)
```

Comment:
so actually turns out a while loop isnt great, it create infinite loops sometimes.... for instance when you hit a NaN